eng/WpfArcadeSdk/tools/CodeGen/AvTrace/GenTraceSources.targets

```diff
@@ -0,0 +1,26 @@
+<Project>
+  <PropertyGroup>
+    <WindowsBaseAvTraceMessageFile>$(WpfSourceDir)\WindowsBase\AvTraceMessages.json</WindowsBaseAvTraceMessageFile>
+    <PresentationCoreAvTraceMessageFile>$(WpfSourceDir)\PresentationCore\AvTraceMessages.json</PresentationCoreAvTraceMessageFile>
+    <PresentationFrameworkAvTraceMessageFile>$(WpfSourceDir)\PresentationFramework\AvTraceMessages.json</PresentationFrameworkAvTraceMessageFile>    
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="$(MSBuildThisFileDirectory)PresentationTraceSources.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>PresentationTraceSources.cs</LastGenOutput>
+      <OutputFilePath>$(WpfSourceDir)\WindowsBase\Generated\</OutputFilePath>
+    </None>
+    <None Update="PresentationTraceSources.cs">
+      <DesignTime>True</DesignTime>
+      <AutoGen>True</AutoGen>
+      <DependentUpon>$(MSBuildThisFileDirectory)PresentationTraceSources.tt</DependentUpon>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <T4ParameterValues Include="AvTraceMessageFiles">
+      <Value>$(WindowsBaseAvTraceMessageFile);$(PresentationCoreAvTraceMessageFile);$(PresentationFrameworkAvTraceMessageFile)</Value>
```

Comment:
Does this target run all at once for three projects, or does this run three times, once each per project? I'm trying to understand why all three of these values have to be embedded here permanently, rather than having these picked up dynamically one at a time.

Reply:
This target only runs once for WindowsBase (it's the only project that includes it).  The GenTraceSources worked in the past was that it built the `PresentationTraceSources` class into WindowsBase only, and not the other assemblies. This is the class, and since it's public i felt it safest to not change this aspect of the design:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.presentationtracesources?view=netframework-4.7.2

I agree that this wasn't ideal, but i also felt this was simple and straightforward enough to not worry about it too much. I'm open to suggestions for improving it if you see something.

Reply:
i do want to write up some documentation relating to this. i'll add some to this PR

Reply:
Another option is that each project opts-in by contributing to an `<ItemGroup>` of `@(AvTraceMessageFiles)`, but only one project _runs_ the target (which, as you say, is `WindowsBase`). At the time the target runs, the accumulated set of config files in `@(AvTraceMessageFiles)` gets processed.